src/components/plots/plot.ts

```diff
@@ -265,28 +265,25 @@ module Plottable {
       extents.set(key, this._datasetKeysInOrder.map((key) => {
         var plotDatasetKey = this._key2PlotDatasetKey.get(key);
         var dataset = plotDatasetKey.dataset;
-        return this._computeExtent(dataset, accScaleBinding.accessor, filter);
+        return this._computeExtent(dataset, accScaleBinding, filter);
       }));
     }
 
-    private _computeExtent(dataset: Dataset, accessor: Accessor<any>, filter: Accessor<boolean>): any[] {
+    private _computeExtent(dataset: Dataset, accScaleBinding: Plots.AccessorScaleBinding<any, any>, filter: Accessor<boolean>): any[] {
+      var accessor = accScaleBinding.accessor;
+      var scale = accScaleBinding.scale;
+
       var data = dataset.data();
       if (filter != null) {
         data = data.filter((d, i) => filter(d, i, dataset));
       }
       var appliedAccessor = (d: any, i: number) => accessor(d, i, dataset);
       var mappedData = data.map(appliedAccessor);
-      if (mappedData.length === 0) {
-        return [];
-      } else if (typeof(mappedData[0]) === "string") {
-        return Utils.Methods.uniq(mappedData);
+
+      if (scale) {
```

Comment:
I think our current style is to use `!= null` for existence checks, although I'm aware of inconsistencies in the codebase.